### PR TITLE
Bugfix clearing stdout stderr on finish & cleanup

### LINKED ISSUE #493: windows openCL compilation issue
hello, I discussed with Rok on Discourse that I was having similar problems compiling openCL models on Windows that he had earlier addressed in Linux.  https://discourse.mc-stan.org/t/opencl-cmdstanr-windows-error/22375/4 

I took his advice to apply these commands:
```
cmdstan_make_local(cpp_options = "CXXFLAGS += -fpermissive")
rebuild_cmdstan(cores = 4)
```

But unfortunately now there is a new error and according to the error message, I am supposed to file a bug report.  So here goes.  Continuing on with the steps described in the Discourse post, I now get this error:

```Compiling Stan program...
In file included from stan/lib/stan_math/stan/math/opencl/prim.hpp:86,
                 from stan/lib/stan_math/stan/math/prim.hpp:7,
                 from stan/src/stan/io/dump.hpp:7,
                 from src/cmdstan/command.hpp:24,
                 from src/cmdstan/main.cpp:1:
stan/lib/stan_math/stan/math/opencl/scalar_type.hpp:14:8: warning: partial specialization of 'struct stan::scalar_type<T, typename std::enable_if<stan::math::conjunction<stan::is_kernel_expression_and_not_scalar<T, void> >::value, void>::type>' after instantiation of 'struct stan::scalar_type<stan::math::constant_<int>, void>' [-fpermissive]
 struct scalar_type<T, require_all_kernel_expressions_and_none_scalar_t<T>> {
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from stan/src/stan/model/model_header.hpp:16:
stan/src/stan/services/util/create_rng.hpp:35:1: internal compiler error: in gt_pch_note_object, at ggc-common.c:276
 }  // namespace stan
 ^
libbacktrace could not find executable to open
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://github.com/r-windows> for instructions.
mingw32-make.exe: *** [make/program:25: stan/src/stan/model/model_header_opencl.hpp.gch] Error 1
Error: An error occured during compilation! See the message above for more information.
```

### LINKED ISSUE #517: optimize() can return failure, even when the underlying fit is succesful
**Describe the bug**
Using `$optimize()` I can reliably (on my computer) get an error message even for the most trivial models, that work when called from command line. It appears there is some sort of race condition as forcing `cmdstanr` to reread the process output once more after the `while (procs$active_procs() == 1)` loop in `.run_other()` finishes makes the fit succeed.

**To Reproduce**

```r
m_test <- cmdstan_model(write_stan_file(r"(
parameters {
  real x;
}
model {
  x ~ normal(0,1);
}
                                        )"))

fit_test <- m_test$optimize()
# Fitting finished unexpectedly! Use the $output() method for more information.

fit_test$output()
# method = optimize
```

**Expected behavior**

A succesful optimization fit.

**Operating system**
Windows 10.

**CmdStanR version number**
0.4.0.9000 (just downloaded from GitHub)

**Additional context**

There is no problem when runnning the model from the command line with the exact same arguments cmdstanr uses. 
In fact, copying and pasting the output processing code

```
  output <- procs$get_proc(id)$read_output_lines()
  procs$process_output(output, id)
  error_output <- procs$get_proc(id)$read_error_lines()
  procs$process_error_output(error_output, id)
```

just below the loop that waits for the process to finish (at https://github.com/stan-dev/cmdstanr/blob/043c8fb991d496229074f51171e7a9198fae343b/R/run.R#L470) makes the problem disappear.

I verified that upon last execution at https://github.com/stan-dev/cmdstanr/blob/043c8fb991d496229074f51171e7a9198fae343b/R/run.R#L464 the `output` only contains one line: `method = optimize`, but once the loop finishes, the output contains the whole result, including the crucial "Optimization terminated normally".

I don't think I understand the code enough to know whether this copy-paste of code is a good solution to the underlying problem, but it lets me sidestep the issue for now.

### PR DESCRIPTION
**Pull request by rok-cesnovar**
#### Summary

Fix #517 by applying the suggestion by @martinmodrak. Fixes #493.  Also a bit of cleanup (see comment).

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting 
(this will be you or your assignee, such as a university or company): 
Rok Češnovar

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)    
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: R/install.R [hunk 1]
```diff
@@ -371,36 +371,6 @@ build_cmdstan <- function(dir,
   )
 }
 
-# Removes files that are used to simplify switching to using threading, opencl or mpi.
-clean_compile_helper_files <- function() {
```
Comment (rok-cesnovar) [inline R/install.R:375]: This has not been needed for some time, we just haven't cleaned it up.

File: R/model.R [hunk 2]
```diff
@@ -457,18 +457,7 @@ compile <- function(quiet = TRUE,
...
-    include_paths <- absolute_path(include_paths)
-    include_paths <- paste0(include_paths, collapse = ",")
-    if (cmdstan_version() >= "2.24") {
-      include_paths_flag <- " --include-paths="
-    } else {
-      include_paths_flag <- " --include_paths="
-    }
-    stancflags_val <- paste0(stancflags_val, include_paths_flag, include_paths, " ")
-  }
+  stancflags_val <- include_paths_stanc3_args(include_paths)
```
Comment (rok-cesnovar) [inline R/model.R:460]: This code is repeated in check_syntax, so I added a utility function with a test.

File: R/utils.R [hunk 3]
```diff
@@ -60,9 +60,7 @@ is_rosetta2 <- function() {
 
 # Returns the type of make command to use to compile depending on the OS
 make_cmd <- function() {
-  # CmdStan 2.21 introduced TBB that requires mingw32-make on Windows
-  ver <- .cmdstanr$VERSION
-  if (os_is_windows() && (is.null(ver) || ver >= "2.21")) {
```
Comment (rok-cesnovar) [inline R/utils.R:65]: This check for versions is not needed. CmdStan 2.20 and earlier will still work with mingw32-make (its just that 2.21 and on requires mingw32-make). The toolchain installation for Windows is made in such a way that users will always have mingw32-make (3.x or 4.x). Not to mention I doubt anyone uses CmdStan 2.20 with CmdStanR :)

File: R/install.R (hunk 1 above)
Comment (jgabry) [reply R/install.R:375]: Awesome, always nice to delete a bunch of code!

**Review by jgabry — APPROVED** [review]
Looks good!

**Pull request merged**